js/preserve-data.js

```diff
@@ -0,0 +1,28 @@
+function saveLocalStorage() {
+  const form = document.querySelector('.form');
+
+  // Add event listener for form submission
+  form.addEventListener('submit', (event) => {
+    event.preventDefault(); // Prevent the form from submitting
+
+    // Get the input values
+    const name = document.getElementById('name').value;
+    const email = document.getElementById('email').value;
+    const message = document.getElementById('message').value;
+
+    // Create an object to store the input data
+    const inputData = {
+      name: name,
+      email: email,
+      message: message,
+    };
+
+    // Save the input data in local storage
+    localStorage.setItem('formData', JSON.stringify(inputData));
+
+    // Submit the form to the server
+    form.submit();
+  });
+}
```

Comment:
Good job so far in storing the form data to LocalStorage.

However:
- [ ] The input field data needs to be stored in localStorage when their content changes even when the form isn't submitted yet. Currently, the data is only stored in localStorage when the form is submitted. 👍🏽
- [ ] The input fields are expected to be populated with the value of the data stored in localStorage for each field. Currently, when the page loads or is refreshed, the input fields are empty and they should contain the data stored in the localStorage. Kindly set the values of the input fields to be the value of each field data stored in localStorage. 👍🏽

**Please refer to the screenshots below**
<img width="1107" alt="Screenshot 2023-05-27 at 5 40 26 PM" src="https://github.com/alyconr/Personal-Portfolio-Project/assets/50941074/57f7bcf8-6020-454b-b443-715fab482e79">